src/app/home/home.component.ts

```diff
@@ -74,6 +74,15 @@ export class HomeComponent implements AfterViewInit, OnDestroy {
             next: () => console.log('pinging server...'),
             error: () => this.disableEditor()
         });
+
+        this.http.get(this.pingURL).subscribe({
+            next: () => console.log('pinging server...'),
+            error: (e: HttpErrorResponse) => {
+                if (e.status === 429) {
+                    this.disableEditorToManyRequests();
+                }
```

Comment:
What happens when not 200 and not 429? Think you should be using `disableEditor` in the else case here.